src/Icicle/Sea/FromAvalanche/Program.hs

```diff
@@ -205,7 +205,7 @@ seaOfXValue v t
       | ArrayT t' <- t
       -> let len = length vs
              writes arr (v',i)
-              = prim (PrimUpdate $ PrimUpdateArrayPut t')
+              = prim (PrimArray $ PrimArrayPutImmutable t')
```

Comment:
no - perhaps I changed it, I can't remember, but we need to convert arrays in case they contain any buffers